content/documentation/internals/nats-server-protocol.md

```diff
@@ -0,0 +1,263 @@
+
+### Example
+
+The following examples concern subject `FOO` which has been assigned an internal subscriber id of `5`, and a client sid of `1`. To unsubscribe from `FOO`:
+
+`UNSUB RSID:5:1\r\n`
+
+To auto-unsubscribe from `FOO` after 5 messages have been received:
+
+`UNSUB RSID:1:1 5\r\n`
+
+## <a name="MSG"></a>MSG
+
+### Syntax
+
+`MSG <subject> <sid> [reply-to] <#bytes>\r\n[payload]\r\n`
+
+where:
+
+* `subject`: Subject name this message was received on
+* `sid`: The unique alphanumeric subscription ID of the subject
+* `reply-to`: The inbox subject on which the publisher is listening for responses
+* `#bytes`: Size of the payload in bytes
+* `payload`: The message payload data
+
+### Description
+
+The `MSG` protocol message delivers a message from another server.
+
+### Example
+
+The following message delivers a message from subject `FOO.BAR`:
+
+`MSG FOO.BAR 9 11\r\nHello World\r\n`
+
+Deliver the same message along with a reply inbox:
+
+`MSG FOO.BAR 9 INBOX.34 11\r\nHello World\r\n`
+
+## <a name="PINGPONG"></a>PING/PONG
+
+### Description
+
+`PING` and `PONG` implement a simple keep-alive mechanism between servers. Once two servers establish a connection with each other, the NATS server will continuously send `PING` messages to other servers at a configurable interval. If another server fails to respond with a `PONG` message within the configured response interval, the server will terminate its connection. If your connection stays idle for too long, it is cut off.
+
+If the another server sends a ping request, a server will reply with a pong message to notify the other server that it is still present.
+
+## <a name="OKERR"></a>+OK/ERR
+
+### Syntax
+
+`+OK`
+
+`-ERR <error message>`
+
+When the `verbose` connection option is set to `true` (the default value), the server acknowledges each well-formed protocol message with a `+OK` message. NATS servers set the `verbose` option to `false` using the [CONNECT](#CONNECT) message
+
+The `-ERR` message is used by the server indicate a protocol, authorization, or other runtime connection error to another server. Most of these errors result in the server closing the connection.
+
+Handling of these errors usually has to be done asynchronously.
+
+Protocol error messages which close the connection:
+
+- `-ERR 'Unknown Protocol Operation'`: Unknown protocol error
+- `-ERR 'Authorization Violation'`: Server failed to authenticate to another server with credentials specified in the [CONNECT](#CONNECT) message.
+- `-ERR 'Authorization Timeout'`: Server took too long to authenticate to another server after establishing a connection (default 1 second)
+- `-ERR 'Parser Error'`: Cannot parse the protocol message sent by another server
+- `-ERR 'Stale Connection'`: PING/PONG interval expired.
+- `-ERR 'Slow Consumer'`: The server pending data size for the connection has reached the maximum size (default 10MB).
```

Comment:
Think @ColinSullivan1 refers to examples such as... https://github.com/nats-io/gnatsd/blob/01b635cc8ffbb4054e84cff927f48ea969bd3396/server/configs/test.conf#L37
(though don't see `MaxPending` implemented check around either?)